Kinvey-Xamarin-Android/Auth/AndroidNativeCredentialStore.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) 2016, Kinvey, Inc. All rights reserved.
+//
+// This software is licensed to you under the Kinvey terms of service located at
+// http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+// software, you hereby accept such terms of service  (and any agreement referenced
+// therein) and agree that you have read, understand and agree to be bound by such
+// terms of service and are of legal age to agree to such terms with Kinvey.
+//
+// This software contains valuable confidential and proprietary information of
+// KINVEY, INC and is subject to applicable licensing agreements.
+// Unauthorized reproduction, transmission or distribution of this file and its
+// contents is a violation of applicable laws.
+
+using System;
+using System.Collections.Generic;
+using Android.Accounts;
+using Android.Content;
+using Newtonsoft.Json;
+
+namespace KinveyXamarin
+{
+	/// <summary>
+	/// Android native credential store.
+	/// </summary>
+	public class AndroidNativeCredentialStore : NativeCredentialStore
+	{
+		private Android.App.Activity appActivity;
+		private Context appContext;
+		private AccountManager accountManager;
+
+		#region NativeStoreCredential implementation
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="T:KinveyXamarin.AndroidNativeCredentialStore"/> class.
+		/// </summary>
+		/// <param name="activity">App activity.</param>
+		public AndroidNativeCredentialStore(Android.App.Activity activity)
+		{
+			appActivity = activity;
+			appContext = appActivity.ApplicationContext;
+			accountManager = AccountManager.Get(appContext);
+		}
+
+		/// <summary>
+		/// Load the specified userID.
+		/// </summary>
+		/// <param name="userID">User identifier.</param>
+		/// <param name="ssoGroupKey">SSO Group Key.</param>
+		override public Credential Load(string userID, string ssoGroupKey)
+		{
+			Credential credential = null;
+
+			try
+			{
+				NativeCredential nc = null;
+
+				var credentials = FindCredentialsForOrg(ssoGroupKey);
+
+				foreach (var c in credentials)
+				{
+					if (userID.Equals(string.Empty) ||
+						userID.Equals(c.UserID))
+					{
+						nc = c;
+						break;
+					}
+				}
+
+				if (nc != null)
+				{
+					credential = Credential.From(nc);
+				}
+			}
+			catch (System.Exception e)
+			{
+				string msg = e.Message;
+			}
+
+			return credential;
+		}
+
+		/// <summary>
+		/// Store the specified userID and credential.
+		/// </summary>
+		/// <param name="userID">User identifier.</param>
+		/// <param name="ssoGroupKey">SSO Group Key.</param>
+		/// <param name="credential">Credential.</param>
+		override public void Store(string userID, string ssoGroupKey, Credential credential)
+		{
+			Dictionary<string, string> properties = new Dictionary<string, string>();
```

Comment:
@tejasranade I will handle this as part of a new ticket I will create, which will be to change the format of the serialized credential to be JSON.